objectserver/indexdb/ecobj.go

```diff
@@ -296,100 +299,158 @@ func (o *ecObject) Replicate(prirep objectserver.PriorityRepJob) error {
 	return err
 }
 
-func (o *ecObject) Stabilize(ring ring.Ring, dev *ring.Device, policy int) error {
-	wg := sync.WaitGroup{}
-	success := true
+func (o *ecObject) nurseryReplicate(rng ring.Ring, partition uint64, nodes []*ring.Device) error {
+	more := rng.GetMoreNodes(partition)
+	var node *ring.Device
+	e := common.NewExpector(o.client)
+	defer e.Cancel()
+	wrs := make([]io.WriteCloser, 0)
+	successReadyCount := 0
+	var responses []*http.Response
+	var ready []bool
+	for i := 0; successReadyCount < nurseryReplicaCount; i++ {
+		if i < len(nodes) {
+			node = nodes[i]
+		} else if node = more.Next(); node == nil {
+			break
+		}
+
+		rp, wp := io.Pipe()
+		defer rp.Close()
+		defer wp.Close()
+		wrs = append(wrs, wp)
+		req, err := http.NewRequest("PUT", fmt.Sprintf("%s://%s:%d/nursery/%s/%s",
+			node.Scheme, node.ReplicationIp, node.ReplicationPort, node.Device, o.Hash), rp)
+		if err != nil {
+			return err
+		}
+		req.Header.Set("X-Backend-Storage-Policy-Index", strconv.Itoa(o.policy))
+		req.Header.Set("Deletion", strconv.FormatBool(o.Deletion))
+		req.Header.Set("Meta-Ec-Scheme", fmt.Sprintf("%d/%d/%d", o.dataFrags, o.parityFrags, o.chunkSize))
```

Comment:
We talked about squeezing the algo in this header a bit ago. Also, do we need to store in the nursery? I guess it doesn't hurt but seems a little odd.